zeroize/derive/src/lib.rs

```diff
@@ -277,16 +315,25 @@ fn derive_zeroize_with_drop(s: synstructure::Structure<'_>) -> TokenStream {
         }
     });
 
+    let zeroize_on_drop_impl = derive_zeroize_on_drop(&s);
+
     let zeroize_impl = derive_zeroize_without_drop(s);
 
     quote! {
         #zeroize_impl
 
         #[doc(hidden)]
         #drop_impl
+
+        #zeroize_on_drop_impl
     }
 }
 
+fn derive_zeroize_on_drop(s: &synstructure::Structure<'_>) -> TokenStream {
```

Comment:
Aah okay, so `derive_zeroize_on_drop` does exist, but only adds the marker trait.

I think it would make more sense to also derive the `Drop` impl here?

Reply:
Can't be done that way because the method body is completely different.

Reply:
Switching the names sounds good to me.

Reply:
Switching the names was quiet confusing, I changed it up a little, how does it look to you now?